<Issue id="40928" key="STL-1660" number="1660" project="10001" reporter="andrew-scantrust" assignee="delventhalz" creator="andrew-scantrust" type="10002" summary="Rest API crashes when no content-type header is found" priority="4" status="10100" created="2019-07-05 07:25:05.0" updated="2019-07-05 07:25:05.0" votes="0" watches="1" workflowId="53893"> <description><! CDATA When posting to the /batches/ endpoint, the API crashes and returns a 500.  It will continue to service requests.  The right response for an unknown/unaccepted content type is 415.  In the case of the /batches/ endpoint, where there is no allowed content type other than application/octet-stream (as far as I know), the content type could be inferred/assumed if missing, or a nice error thrown saying what the user should do ("Please add the header ...")       {code:java} saw.api |  2019-07-05 07:08:18.062 ERROR web_protocol  Error handling request saw.api | Traceback (most recent call last): saw.api | File "/usr/lib/python3/dist-packages/aiohttp/web_protocol.py", line 416, in start saw.api | resp = yield from self._request_handler(request) saw.api | File "/usr/lib/python3/dist-packages/aiohttp/web.py", line 323, in _handle saw.api | resp = yield from handler(request) saw.api | File "/usr/lib/python3/dist-packages/sawtooth_rest_api/route_handlers.py", line 139, in submit_batches saw.api | if request.headers 'Content-Type'  != 'application/octet-stream': saw.api | File "multidict/_multidict.pyx", line 124, in multidict._multidict._Base.__getitem__ (multidict/_multidict.c:3486) saw.api | File "multidict/_multidict.pyx", line 119, in multidict._multidict._Base._getone (multidict/_multidict.c:3421) saw.api | KeyError: "Key not found: 'Content-Type'" {code}       ></description> </Issue>
